From: =?utf-8?q?Jonas_Dre=C3=9Fler?= <verdre@v0yd.nl>
Date: Wed, 16 Feb 2022 20:04:03 +0100
Subject: dash: Subtract vertical margins from availHeight

The vertical margins are part of this._maxHeight, so we need to subtract
it in order to propery limit icon sizes when the available height is too
small.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2191>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/2607880bf6
---
 js/ui/dash.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/ui/dash.js b/js/ui/dash.js
index 3111174..7c449f8 100644
--- a/js/ui/dash.js
+++ b/js/ui/dash.js
@@ -610,6 +610,7 @@ var Dash = GObject.registerClass({
                        (iconChildren.length - 1) * spacing;
 
         let availHeight = this._maxHeight;
+        availHeight -= this.margin_top + this.margin_bottom;
         availHeight -= this._background.get_theme_node().get_vertical_padding();
         availHeight -= themeNode.get_vertical_padding();
         availHeight -= buttonHeight - iconHeight;
